### ISSUE TITLE
Does llama3-70b support FSDP or DDP

### ISSUE THREAD

**Issue by BenYyyyyy** [meta-llama/llama3#241]
The llama3-70b model loads weights on 8 GPUs, and it seems like a FSDP method. I tried to use the `torch.distributed.fsdp.FullyShardedDataParallel` to wrap my model which uses llama3-70b as the text generator.  However, I found it does work. The code is like that:
```
llama_model = my_llama.build(
                ckpt_dir = ckpt_dir,
                tokenizer_path =tokenizer_path,
                max_seq_len = max_seq_len,
                max_batch_size = max_batch_size,
                model_parallel_size = model_parallel_size) 
    
self.llama_model = FSDP(llama_model)
```

And I also use `torch.utils.data.distributed.DistributedSampler` to build the sampler in dataloader:

```
self.train_sampler = DistributedSampler(train_dataset, shuffle=True)
self.train_dataloader = DataLoader(train_dataset, 
                                           batch_size=batch_size, 
                                           #generator=torch.Generator(device='cuda'),
                                           num_workers=1, 
                                           pin_memory=True,
                                           shuffle=(self.train_sampler is None),
                                           sampler=self.train_sampler,
                                           )
```

And I got error:
```
[rank1]:[E ProcessGroupNCCL.cpp:563] [Rank 1] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=1, OpType=ALLREDUCE, NumelIn=704512, NumelOut=704512, Timeout(ms)=600000) ran for 600013 milliseconds before timing out.
[rank0]:[E ProcessGroupNCCL.cpp:563] 
```

My other codes should be correct and they work well when I do not use FSDP.  The reason why I try to use FSDP is because I found the 8 ranks seems return the same loss and result; I record the result and loss in wandb and found the 8 devices have the same output. Hence I tried to implement the data parallel to utilize the 8 GPUs.

Can you tell me how to run llama3-70b with FSDP or DDP?

**Comment by awgu** [comment]
Maybe https://github.com/pytorch/torchtitan could be a useful reference?

**Comment by subramen** [comment]
+1 to torchtitan. We also have a guide here https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/finetuning/multigpu_finetuning.md 

Closing this issue, please reopen if you need any clarification!
